test/widgets/dialog_checks.dart

```diff
@@ -6,21 +8,41 @@ import 'package:flutter_test/flutter_test.dart';
 /// Checks for an error dialog matching an expected title
 /// and, optionally, matching an expected message. Fails if none is found.
 ///
-/// On success, returns the widget's "OK" button.
+/// On success, returns the widget's "OK" button
+/// (which is a [CupertinoDialogAction] for OS platforms).
+/// 
 /// Dismiss the dialog by calling `tester.tap(find.byWidget(okButton))`.
 Widget checkErrorDialog(WidgetTester tester, {
   required String expectedTitle,
   String? expectedMessage,
 }) {
-  final dialog = tester.widget<AlertDialog>(find.byType(AlertDialog));
-  tester.widget(find.descendant(matchRoot: true,
-    of: find.byWidget(dialog.title!), matching: find.text(expectedTitle)));
-  if (expectedMessage != null) {
+  if (defaultTargetPlatform == TargetPlatform.iOS 
+    || defaultTargetPlatform == TargetPlatform.macOS) {
+        
+    final dialog = tester.widget<CupertinoAlertDialog>(find.byType(CupertinoAlertDialog));
+
     tester.widget(find.descendant(matchRoot: true,
-      of: find.byWidget(dialog.content!), matching: find.text(expectedMessage)));
-  }
+    of: find.byWidget(dialog.title!), matching: find.text(expectedTitle)));
 
-  return tester.widget(
-    find.descendant(of: find.byWidget(dialog),
-      matching: find.widgetWithText(TextButton, 'OK')));
+    if (expectedMessage != null) {
+      tester.widget(find.descendant(matchRoot: true,
+        of: find.byWidget(dialog.content!), matching: find.text(expectedMessage)));
+    }
+
+    return tester.widget(
+      find.descendant(of: find.byWidget(dialog),
+        matching: find.widgetWithText(CupertinoDialogAction, 'OK')));
+
+  } 
+  else {
+    final dialog = tester.widget<Dialog>(find.byType(Dialog));
+    tester.widget(find.widgetWithText(Dialog, expectedTitle));
+    if (expectedMessage != null) {
+      tester.widget(find.widgetWithText(Dialog, expectedMessage));
+    }
+    return tester.widget(
+      find.descendant(of: find.byWidget(dialog),
+        matching: find.widgetWithText(TextButton, 'OK')));
+  }
```

Comment:
Let's use an exhaustive `switch` on `defaultTargetPlatform`, like we do elsewhere. Also, there are several formatting nits that makes this code harder to read than it needs to be.

Proposal:

```dart
  switch (defaultTargetPlatform) {
    case TargetPlatform.android:
    case TargetPlatform.fuchsia:
    case TargetPlatform.linux:
    case TargetPlatform.windows: {
      final dialog = tester.widget<Dialog>(find.byType(Dialog));
      tester.widget(find.widgetWithText(Dialog, expectedTitle));
      if (expectedMessage != null) {
        tester.widget(find.widgetWithText(Dialog, expectedMessage));
      }
      return tester.widget(
        find.descendant(of: find.byWidget(dialog),
          matching: find.widgetWithText(TextButton, 'OK')));
    }
    case TargetPlatform.iOS:
    case TargetPlatform.macOS: {
      final dialog = tester.widget<CupertinoAlertDialog>(
        find.byType(CupertinoAlertDialog));
      tester.widget(find.descendant(matchRoot: true,
        of: find.byWidget(dialog.title!), matching: find.text(expectedTitle)));
      if (expectedMessage != null) {
        tester.widget(find.descendant(matchRoot: true,
          of: find.byWidget(dialog.content!), matching: find.text(expectedMessage)));
      }
      return tester.widget(find.descendant(of: find.byWidget(dialog),
        matching: find.widgetWithText(CupertinoDialogAction, 'OK')));
    }
  }
```